.pre-commit-config.yaml

```diff
@@ -0,0 +1,42 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: trailing-whitespace
+      - id: mixed-line-ending
+        args:
+          - --fix=lf
+      - id: end-of-file-fixer
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.5.1
+    hooks:
+      - id: prettier
+        types_or:
+          - markdown
+          - toml
+          - yaml
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py37-plus
+
+  - repo: https://github.com/omnilib/ufmt
+    rev: v1.3.1
+    hooks:
+      - id: ufmt
+        additional_dependencies:
+          - black == 21.12b0
+          - usort == 1.0.1
+
+  - repo: https://github.com/pycqa/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+        additional_dependencies:
+          - flake8-docstrings == 1.6.0
```

Comment:
`setup.cfg` contained configuration for `pydocstyle`, which can also be run through this plugin. What I didn't find is any invocation of it. If that is intended, I can remove the plugin here.

Reply:
I think we can remove it. Couldn't find any invocation either.

Reply:
Found the usage: https://github.com/pytorch/text/blob/1f17c0ae8f65478148af62b6807d1b8f45dbdecb/.circleci/config.yml#L594-L609. Do you want me to kill this workflow and simply re-add the plugin?

Reply:
> . Do you want me to kill this workflow and simply re-add the plugin?

Yeah, let's do that. Consolidating all of this in pre-commit seems better than the existing workflow!